bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.kt

```diff
@@ -0,0 +1,330 @@
+
+    /**
+     * Set for which releaseStages errors should be sent to Bugsnag.
+     * Use this to stop errors from development builds being sent.
+     *
+     * For example:
+     *
+     * client.setEnabledReleaseStages("production");
+     */
+    var enabledReleaseStages: Set<String> = emptySet()
+
+    var enabledBreadcrumbTypes: Set<BreadcrumbType> = BreadcrumbType.values().toSet()
+
+    /**
+     * Set which packages should be considered part of your application.
+     * Bugsnag uses this to help with error grouping, and stacktrace display.
+     *
+     * For example:
+     *
+     * client.setProjectPackages("com.example.myapp");
+     *
+     * By default, we'll mark the current package name as part of you app.
+     */
+    var projectPackages: Set<String> = emptySet()
+
+    /**
+     * Add a "on error" callback, to execute code at the point where an error report is
+     * captured in Bugsnag.
+     *
+     * You can use this to add or modify information attached to an Event
+     * before it is sent to your dashboard. You can also return
+     * `false` from any callback to prevent delivery. "on error"
+     * callbacks do not run before reports generated in the event
+     * of immediate app termination from crashes in C/C++ code.
+     *
+     * For example:
+     *
+     * Bugsnag.addOnError(new OnError() {
+     * public boolean run(Event event) {
+     * event.setSeverity(Severity.INFO);
+     * return true;
+     * }
+     * })
+     *
+     * @param onError a callback to run before sending errors to Bugsnag
+     * @see OnError
+     */
+    override fun addOnError(onError: OnError) {
+        if (!onErrorTasks.contains(onError)) {
+            onErrorTasks.add(onError)
+        }
+    }
+
+    override fun removeOnError(onError: OnError) {
+        onErrorTasks.remove(onError)
+    }
+
+    /**
+     * Adds an on breadcrumb callback
+     *
+     * @param onBreadcrumb the on breadcrumb callback
+     */
+    override fun addOnBreadcrumb(onBreadcrumb: OnBreadcrumb) {
+        if (!breadcrumbCallbacks.contains(onBreadcrumb)) {
+            breadcrumbCallbacks.add(onBreadcrumb)
+        }
+    }
+
+    /**
+     * Removes an on breadcrumb callback
+     *
+     * @param onBreadcrumb the on breadcrumb callback
+     */
+    override fun removeOnBreadcrumb(onBreadcrumb: OnBreadcrumb) {
+        breadcrumbCallbacks.remove(onBreadcrumb)
+    }
+
+    /**
+     * Adds an on session callback
+     *
+     * @param onSession the on session callback
+     */
+    override fun addOnSession(onSession: OnSession) {
+        if (!sessionCallbacks.contains(onSession)) {
+            sessionCallbacks.add(onSession)
+        }
+    }
+
+    /**
+     * Removes an on session callback
+     *
+     * @param onSession the on session callback
+     */
+    override fun removeOnSession(onSession: OnSession) {
+        sessionCallbacks.remove(onSession)
+    }
+
+    override fun addMetadata(section: String, value: Any?) =
```

Comment:
I think the type of `value` can be more specific here – it needs to be a `Map<String, any>`

Reply:
good catch - I've updated the implementation to use `Map<String, Any?>`